staging/src/k8s.io/component-helpers/node/util/ips.go

```diff
@@ -52,20 +51,20 @@ func parseNodeIP(nodeIP string, allowDual, sloppy bool) ([]net.IP, error) {
 	}
 
 	if len(nodeIPs) > 2 || (len(nodeIPs) == 2 && netutils.IsIPv6(nodeIPs[0]) == netutils.IsIPv6(nodeIPs[1])) {
-		return nil, fmt.Errorf("must contain either a single IP or a dual-stack pair of IPs")
+		return nil, invalidIPs, fmt.Errorf("must contain either a single IP or a dual-stack pair of IPs")
 	} else if len(nodeIPs) == 2 && !allowDual {
-		return nil, fmt.Errorf("dual-stack not supported in this configuration")
+		return nil, invalidIPs, fmt.Errorf("dual-stack not supported in this configuration")
 	} else if len(nodeIPs) == 2 && (nodeIPs[0].IsUnspecified() || nodeIPs[1].IsUnspecified()) {
-		return nil, fmt.Errorf("dual-stack node IP cannot include '0.0.0.0' or '::'")
+		return nil, invalidIPs, fmt.Errorf("dual-stack node IP cannot include '0.0.0.0' or '::'")
 	}
 
-	return nodeIPs, nil
+	return nodeIPs, invalidIPs, nil
 }
 
 // ParseNodeIPArgument parses kubelet's --node-ip argument. If nodeIP contains invalid
 // values, they will be logged and ignored. Dual-stack node IPs are allowed if
 // cloudProvider is unset, or if it is `"external"` and allowCloudDualStack is true.
-func ParseNodeIPArgument(nodeIP, cloudProvider string, allowCloudDualStack bool) ([]net.IP, error) {
+func ParseNodeIPArgument(nodeIP, cloudProvider string, allowCloudDualStack bool) ([]net.IP, []string, error) {
```

Comment:
The new return value needs to be explained, replacing the "will be logged and ignored". Something like:

> If nodeIP contains invalid values, they will be returned as strings. This is done also when an error is returned. The caller then can decide what to do with the invalid values.